platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/deduplication/EventDeduplicator.java

```diff
@@ -107,12 +109,19 @@ public EventDeduplicator(
         this.eventIntakeMetrics = Objects.requireNonNull(eventIntakeMetrics);
 
         this.disparateSignatureAccumulator = platformContext.getMetrics().getOrCreate(DISPARATE_SIGNATURE_CONFIG);
+        if (platformContext.getConfiguration().getConfigData(EventConfig.class).getAncientMode()
+                == AncientMode.BIRTH_ROUND_THRESHOLD) {
+            nonAncientEventWindow = NonAncientEventWindow.INITIAL_EVENT_WINDOW_BIRTH_ROUND;
```

Comment:
Optionally, you could create a static method in `NonAncientEventWindow` which creates an initial window from the enum value, since this chunk of code is repeated in lots of places

Reply:
suggestion adopted, things are much cleaner now, thank you.